commons-impl/src/main/java/com/connexta/transformation/commons/inmemory/status/InMemoryMetadataTransformation.java

```diff
@@ -0,0 +1,178 @@
+
+import com.connexta.transformation.commons.api.ErrorCode;
+import com.connexta.transformation.commons.api.RequestInfo;
+import com.connexta.transformation.commons.api.exceptions.TransformationException;
+import com.connexta.transformation.commons.api.status.MetadataTransformation;
+import com.google.common.io.ByteSource;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.OptionalLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InMemoryMetadataTransformation implements MetadataTransformation {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(InMemoryMetadataTransformation.class);
+  private final Object stateLock = new Object();
+  private final String metadataType;
+  private final String transformId;
+  private final RequestInfo requestInfo;
+  private final Instant startTime;
+  private Instant completionTime;
+  private byte[] content;
+  private String contentType;
+  private State state;
+  private ErrorCode failureReason;
+  private String failureMessage;
+
+  public InMemoryMetadataTransformation(
+      String metadataType, String transformId, RequestInfo requestInfo) {
+    this.startTime = Instant.now();
+    this.state = State.IN_PROGRESS;
+    this.metadataType = metadataType;
+    this.transformId = transformId;
+    this.requestInfo = requestInfo;
+  }
+
+  @Override
+  public String getMetadataType() {
+    return metadataType;
+  }
+
+  @Override
+  public Optional<InputStream> getContent() {
+    if (content == null) {
+      return Optional.empty();
+    } else {
+      try {
+        return Optional.of(ByteSource.wrap(content).openStream());
+      } catch (IOException e) {
+        LOGGER.error(
+            "Unable to read contents of [{}] metadata for transformation [{}].",
+            metadataType,
+            transformId);
+        return Optional.empty();
+      }
+    }
+  }
+
+  @Override
+  public Optional<String> getContentType() {
+    return Optional.ofNullable(contentType);
+  }
+
+  @Override
+  public OptionalLong getContentLength() {
+    if (content == null) {
+      return OptionalLong.empty();
+    } else {
+      return OptionalLong.of(content.length);
+    }
+  }
+
+  @Override
+  public void succeed(String contentType, InputStream contentStream)
+      throws TransformationException, IOException, IllegalStateException {
+    Instant completionTime = Instant.now();
+    synchronized (stateLock) {
+      checkForCompletion();
+      try {
+        content = ByteStreams.toByteArray(contentStream);
+      } finally {
+        try {
+          contentStream.close();
+        } catch (IOException e) {
+          LOGGER.debug(
+              "Unable to close contents stream of [{}] metadata for transformation [{}].",
+              metadataType,
+              transformId);
```

Comment:
🎉 Thank you for not failing the whole thing since we already read all the bytes we needed from the stream anyway